apps/api/v2/src/modules/oauth/user/oauth-user.controller.ts

```diff
@@ -0,0 +1,93 @@
+import { AccessTokenGuard } from "@/modules/auth/guard/access-token/access-token.guard";
+import { OAuthClientGuard } from "@/modules/oauth/guard/oauth-client/oauth-client.guard";
+import { CreateUserInput } from "@/modules/user/input/create-user";
+import { UpdateUserInput } from "@/modules/user/input/update-user";
+import { UserRepository } from "@/modules/user/user.repository";
+import {
+  Body,
+  Controller,
+  Post,
+  Logger,
+  UseGuards,
+  Get,
+  HttpCode,
+  HttpStatus,
+  NotFoundException,
+  Param,
+  Put,
+  BadRequestException,
+} from "@nestjs/common";
+import { User } from "@prisma/client";
+
+import { DUPLICATE_RESOURCE, SUCCESS_STATUS } from "@calcom/platform-constants";
+import { ApiResponse } from "@calcom/platform-types";
+
+@Controller({
+  path: "oauth-clients/:clientId/users",
+  version: "2",
+})
+export class OAuthUserController {
+  private readonly logger = new Logger("UserController");
+
+  // TODO: Inject service for access and refresh tokens
+  constructor(private readonly userRepository: UserRepository) {}
+
+  @Post("/")
+  @UseGuards(OAuthClientGuard)
+  async createUser(
+    @Param("clientId") oAuthClientId: string,
+    @Body() body: CreateUserInput
+  ): Promise<ApiResponse<CreateUserResponse>> {
+    this.logger.log(
+      `Creating user with data: ${JSON.stringify(body, null, 2)} for OAuth Client with ID ${oAuthClientId}`
+    );
+
+    const exists = await this.userRepository.findByEmail(body.email);
+
+    if (exists) {
+      throw new BadRequestException(DUPLICATE_RESOURCE);
+    }
```

Comment:
```suggestion
    const existingUser = await this.userRepository.findByEmail(body.email);

    if (Boolean(existingUser)) {
      throw new BadRequestException(DUPLICATE_RESOURCE);
    }
```
suggestion for more semanticity

Reply:
I agree with renaming "exists" to "existingUser" but I did not add "Boolean" in [refactor: variable naming](https://github.com/calcom/cal.com/pull/12853/commits/3ca77178df3f7e70a056c419b60b483d2fce2c5d), because in my opinion its redundant - if "findByEmail" returns an object or null, then anything other than null in this case is true. It would be similar to explicitly writing types in typescript IMO.